### PR TITLE
split key default to false

### LINKED ISSUE #41: Shutting down with log
# Background
The current log outputs for `shutting down...` has no enough information whether its shutting down due to given signal or other reasons. The line must have more information so that the logs themselves are enough to understand which is responsible for shutting the sia.

## TODOs
- [x] Improve the `shutting down...` log with the cause 
  - https://github.com/AthenZ/k8s-athenz-sia/blob/c8478297a9d228ffc0a6a1ea469ad0ef8a682dc8/cmd/athenz-sia/main.go#L87-L90

### LINKED ISSUE #42: Implementation for ROLE_CERT_KEY_FILE_OUTPUT and its logic happens to be destructive.
# Background
Outputting role cert key for role certificates feature has been implemented in https://github.com/AthenZ/k8s-athenz-sia/pull/38 but  the default value for `ROLE_CERT_KEY_FILE_OUTPUT` is `true`, which it should be `false`

Also, it is not really straightforward for the code below and I believe it should simply output/not output based on the config itself.
https://github.com/AthenZ/k8s-athenz-sia/blob/1c66fbc53936d9007dd6adecc88e5dba130ce5c3/pkg/identity/certificated.go#L107-L113

## TODOs
- [x] Change the default `ROLE_CERT_KEY_FILE_OUTPUT` from `true` to `false`
- [x] Change the code [here](https://github.com/AthenZ/k8s-athenz-sia/blob/1c66fbc53936d9007dd6adecc88e5dba130ce5c3/pkg/identity/certificated.go#L107-L113) as the following instead.
  - [ ] Make sure the appropriate logger is handled too for [this](https://github.com/AthenZ/k8s-athenz-sia/issues/42#issuecomment-1774453993).
```go
// Much simpler and straightforward. (Does not require to understand `id == nil`)
if idConfig.RoleCertKeyFileOutput {
  outKeyPath := filepath.Join(idConfig.RoleCertDir, rolecert.Domain+idConfig.RoleCertFilenameDelimiter+rolecert.Role+".key.pem")
  log.Debugf("Saving x509 role cert key[%d bytes] at [%s]", len(roleKeyPEM), outKeyPath)
  if err := w.AddBytes(outKeyPath, 0644, roleKeyPEM); err != nil {
    return errors.Wrap(err, "unable to save x509 role cert key")
  }
}
```
- [x] [Env](https://github.com/AthenZ/k8s-athenz-sia/blob/c8478297a9d228ffc0a6a1ea469ad0ef8a682dc8/athenz-sia.env) is modified accordingly
- [x] ~Add the env for the [Makefile](https://github.com/AthenZ/k8s-athenz-sia/blob/main/Makefile#L21-L53) too~
  - will be handled here instead in https://github.com/AthenZ/k8s-athenz-sia/issues/47

### PR DESCRIPTION
**Pull request by WindzCUHK**
# Description

- `ROLE_CERT_KEY_FILE_OUTPUT` default value: `true` => `false`
- warn user for problematic cases
  ```
  WARNING[2023-10-25T16:01:46+09:00] Rotating KEY_FILE[./my.private.key.pem] may cause key mismatch with issued role certificate due to different rotation cycle. Please manually restart SIA when you rotate the key file.
  ```
- improve shutdown msg
  ```
  ^C
  INFO[2023-10-26T10:06:10+09:00] Initiating shutdown with received signal interrupt ...
  INFO[2023-10-26T10:06:10+09:00] Initiating shutdown of certificate provider daemon ...
  INFO[2023-10-26T10:06:10+09:00] Initiating shutdown of token provider daemon ...
  INFO[2023-10-26T10:06:10+09:00] Initiating shutdown of metrics exporter daemon ...
  INFO[2023-10-26T10:06:10+09:00] Shutdown completed!
  ```

## Type of change

- [x] Bug fix
- [ ] New feature
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Non-code changes (update documentation, pipeline, etc.)

### Flags

- [ ] Breaks backward compatibility
- [ ] Requires a documentation update
- [ ] Has untestable code

## Related issue/PR

- Fixes #42
- Fixes #41

---

## Checklist

- [x] Followed the guidelines in the CONTRIBUTING document
- [x] Added prefix `[skip ci]`/`[ci skip]`/`[no ci]`/`[skip actions]`/`[actions skip]` in the PR title if necessary
- [x] Tested and linted the code
- [x] Commented the code
- [x] Made corresponding changes to the documentation
- [ ] Passed all pipeline checking

## Checklist for maintainer

- Use `Squash and merge`
- Double-confirm the merge message has prefix `[skip ci]`/`[ci skip]`/`[no ci]`/`[skip actions]`/`[actions skip]`
- Delete the branch after merge

### CODE REVIEW

File: cmd/athenz-sia/main.go [hunk 1]
```diff
@@ -85,8 +85,8 @@ func main() {
 	}
 
 	if !idConfig.Init {
-		<-ch // wait until receiving os.Signal from channel ch
-		log.Println("Shutting down...")
+		s := <-ch // wait until receiving os.Signal from channel ch
+		log.Printf("Receive %s signal, will shut down...\n", s.String())
```
Comment (mlajkim) [inline cmd/athenz-sia/main.go:89]:
```suggestion
		log.Printf("Initiating shutdown with received signal %s ...\n", s.String())
```

File: pkg/identity/certificated.go [hunk 2]
```diff
@@ -354,7 +353,7 @@ func Certificated(idConfig *config.IdentityConfig, stopChan <-chan struct{}) (er
 					log.Errorf("Failed to refresh x509 certificate after multiple retries: %s", err.Error())
 				}
 			case <-stopChan:
-				log.Info("Certificate provider will shutdown")
+				log.Info("Certificate provider will shut down...")
```
Comment (mlajkim) [inline pkg/identity/certificated.go:356]:
```suggestion
				log.Info("Initiating shutdown of certificate provider daemon ...")
```

File: pkg/identity/healthcheckd.go [hunk 3]
```diff
@@ -59,7 +59,7 @@ func Healthcheckd(idConfig *config.IdentityConfig, stopChan <-chan struct{}) (er
 		defer close(shutdownChan)
 
 		<-stopChan
-		log.Info("Health check server will shutdown")
+		log.Info("Health check server will shut down...")
```
Comment (mlajkim) [inline pkg/identity/healthcheckd.go:62]:
```suggestion
		log.Info("Initiating shutdown of health check daemon ...")
```

File: pkg/identity/metricsd.go [hunk 4]
```diff
@@ -98,7 +98,7 @@ func Metricsd(idConfig *config.IdentityConfig, stopChan <-chan struct{}) (error,
 		defer close(shutdownChan)
 
 		<-stopChan
-		log.Info("Metrics exporter will shutdown")
+		log.Info("Metrics exporter will shut down...")
```
Comment (mlajkim) [inline pkg/identity/metricsd.go:101]:
```suggestion
		log.Info("Initiating shutdown of metrics exporter daemon ...")
```

File: pkg/token/daemon.go [hunk 5]
```diff
@@ -274,7 +274,7 @@ func Tokend(idConfig *config.IdentityConfig, stopChan <-chan struct{}) (error, <
 					log.Errorf("Failed to refresh tokens after multiple retries: %s", err.Error())
 				}
 			case <-stopChan:
-				log.Info("Token provider will shutdown")
+				log.Info("Token provider will shut down...")
```
Comment (mlajkim) [inline pkg/token/daemon.go:277]:
```suggestion
				log.Info("Initiating shutdown of token provider daemon ...")
```

**Review by mlajkim — COMMENTED** [review]
First review done.
No Makefile? @WindzCUHK

**Comment by WindzCUHK** [comment]
@mlajkim 
> No Makefile?
Replied in https://github.com/AthenZ/k8s-athenz-sia/issues/42#issuecomment-1778644098
Can we skip it in this PR, and continue the discussion in https://github.com/AthenZ/k8s-athenz-sia/issues/47?

**Review by mlajkim — APPROVED** [review]
lgtm

**Pull request merged**